### PR TITLE
Modified crystal init to check for pre-existing files and folders.

### LINKED ISSUE #1472: Trying to create a lib called First
Hi dear folks,

first of all - this is a very promising project. Thanks for keeping us from writing C code ;-).

This error occurred on my very first shoots. I already created a file called `first.cr` and compiled it. Then later I tried to create a lib called `First` (uppercase!)

    ~/p/crystal ❯❯❯ ls -l                                                                                                    ⏎
    total 632
    drwxr-xr-x   5 andwen  staff     170 14 Sep 20:28 .
    drwxr-xr-x  72 andwen  staff    2448 14 Sep 20:24 ..
    drwxr-xr-x   3 andwen  staff     102 14 Sep 20:27 .crystal
    -rwxr-xr-x   1 andwen  staff  318692 14 Sep 20:28 first
    -rw-r--r--   1 andwen  staff      74 14 Sep 20:26 first.cr
    ~/p/crystal ❯❯❯ crystal init lib First
    Unable to create directory './First': File exists
    *Errno::new<String>:Errno +284 [4445832476]
    *Dir::mkdir_p<String>:Int32 +536 [4448080536]
    *Crystal::Command::run<Array(String)>:(Nil | Bool | FileDescriptorIO+ | Crystal::Compiler::Result | Array(Crystal::ImplementationTrace) | Array(Crystal::Init::View+:Class)) +2272 [4451654224]
    main +17325 [4451651277]

    Error: you've found a bug in the Crystal compiler. Please open an issue, including source code that will allow us to reproduce the bug: https://github.com/manastech/crystal/issues
    
I hope this helps a bit ...

Cheers

Andy

### PR DESCRIPTION
**Pull request by vjdhama**
- fixes #1472 & #1555

### CODE REVIEW

File: src/compiler/crystal/tools/init.cr [hunk 1]
```diff
@@ -56,6 +56,25 @@ DIR  - directory where project will be generated,
       github_user.empty? ? default : github_user
     end
 
+    def self.fetch_name(opts, args)
+      name = fetch_required_parameter(opts, args, "NAME")
+      name_type = fetch_name_type(name)
+      if name_type
+        puts "invalid NAME value: #{name} #{name_type} already exists"
```
Comment (jhass) [inline src/compiler/crystal/tools/init.cr]: "foo directory/file already exists", doesn't read that backwards? Also what about printing to stderr?

**Comment by jhass** [comment]
Specs? :)

File: src/compiler/crystal/tools/init.cr (hunk 1 above)
Comment (vjdhama) [inline src/compiler/crystal/tools/init.cr]: I hadn't thought too much about this message. What would you suggest to be a good error message? 

Secondly for printing as an error, I'll change it to `STDERR.puts`.

Comment (kirbyfan64) [inline src/compiler/crystal/tools/init.cr]: "directory/file 'foo' already exists"

**Comment by vjdhama** [comment]
I am adding specs right now.

File: src/compiler/crystal/tools/init.cr [hunk 2]
```diff
@@ -56,6 +56,18 @@ DIR  - directory where project will be generated,
       github_user.empty? ? default : github_user
     end
 
+    def self.fetch_name(opts, args)
+      name = fetch_required_parameter(opts, args, "NAME")
+      puts "Path : #{name}"
+      puts Dir.exists?(name)
```
Comment (jhass) [inline src/compiler/crystal/tools/init.cr]: Leftover debug statements?

Comment (vjdhama) [inline src/compiler/crystal/tools/init.cr]: Fixed.

File: src/compiler/crystal/tools/init.cr [hunk 3]
```diff
@@ -56,6 +56,16 @@ DIR  - directory where project will be generated,
       github_user.empty? ? default : github_user
     end
 
+    def self.fetch_name(opts, args)
+      name = fetch_required_parameter(opts, args, "NAME")
+      if Dir.exists?(name) || File.exists?(name)
+        puts "file or directory #{name} already exists"
+        puts opts
```
Comment (jhass) [inline src/compiler/crystal/tools/init.cr]: Mmh, sure about printing opts here? What for?

Comment (vjdhama) [inline src/compiler/crystal/tools/init.cr]: Yeah I agree. That's not required. I updated the PR.

**Comment by jhass** [comment]
Thanks!

**Pull request merged**